week2/MAKEME.md

```diff
@@ -0,0 +1,112 @@
+
+**Exercise 2: Dog photo gallery**
+
+Let's make a randomized dog photo gallery!
+
+In this exercise we'll be using the following API endpoint: `https://dog.ceo/api/breeds/image/random`
+
+1. Create 3 functional components: `<DogGallery>`, `<DogPhoto>` and `<Button>`
+2. Inside `<DogGallery>` create a state variable called `dogPhotos` (initialize with value `[]`) and state handler `setDogPhotos`
+3. Inside (before the return statement) also create a function called `getDogPhoto`. The purpose of this function is to make an API call to `https://dog.ceo/api/breeds/image/random`. You can either use `fetch` or `axios`. At the end of the function push the new `dog image URL` into the `dogPhotos` array state variable using `setDogPhotos`
+4. Inside the `<Button>` component, create a `<button>` that has the text "Get a dog!" and `onClick` attribute
+5. Pass down the function `getDogPhoto` to `<Button>`
+6. Inside `<DogGallery>` return all the dogs stored in the `dogPhotos` array using the `map()` function. Pass down each `dogPhoto` into an instance of `<DogPhoto>`.
+7. However, when there are no dogs in the array yet make sure to display the message "Get your first dog by clicking the button!"
+8. Test your components by importing it into the top level component, which is `<App>`
+
+**Exercise 3: Random Joke Generator**
+
+Never run out of jokes with the Random Joke Generator!
+
+In this exercise we'll be using the following API endpoint: `https://official-joke-api.appspot.com/random_joke`
+
+1. Create 2 functional components: `<RandomJoke>` and `<Joke>`
+2. Inside `<RandomJoke>` create a state variable called `joke` (with initial state `{}`) and state handler `setJoke`
+3. Make use of the `useEffect` hook to make an asynchronous API call to the API endpoint. Put the end result into the state variable using `setJoke`
+4. Now pass the reassigned state variable `joke` to the `<Joke>` component
+5. Inside the `<Joke>` component, display the `setup` and `punchline` attributes in 2 `<p>` tags
+6. Test out your work by importing your components into the top level component, which is `<App>`
+
+## **3. Code along**
+
+In the following tutorial you'll learn how to
+
+- [Build a Calculator with React and Flexbox](https://www.youtube.com/watch?v=KzYUuTiHdiY)
+
+## **4. PROJECT: Hack Your Weather II**
+
+> Make sure you're building on last week's codebase.
+
+Let's continue building on our application! This week you'll replace the application's data source: instead of using a JSON file you'll be using an external API, called [OpenWeatherMap](https://openweathermap.org/). In addition to that, you'll add a `search functionality` that allows a user to type in a city name and retrieve weather data on it!
+
+### 1. Setup
+
+1. Sign up for an API key at [OpenWeatherMap App ID](https://openweathermap.org/appid)
+2. Create a `.env` file in the root of your project folder
+3. Paste the API key inside the `.env` file with the name `OPENWEATHERMAP_API_KEY`
```

Comment:
correction: `REACT_APP_OPENWEATHERMAP_API_KEY`
cra requires env keys to be appended with `REACT_APP`

Reply:
Good point. I'll change it.